src/components/Menu/index.tsx

```diff
@@ -1,12 +1,23 @@
+'use client'
+
 import { Bug, Cog, Files, FlaskConical, GitFork, Monitor, Puzzle, Search } from 'lucide-react'
+
 import { MenuButton } from './MenuButton'
 import { SocialLinksMenuButton } from './SocialLinksMenuButton'
+import { isExplorerOpenAtom } from '@/store/explorerStore'
+import { useAtom, useSetAtom } from 'jotai'
 
 export function Menu() {
+  const setIsExplorerOpen = useSetAtom(isExplorerOpenAtom)
+
+  function toggleExplorerVisibility() {
+    setIsExplorerOpen((state) => !state)
+  }
+
   return (
-    <div className="flex flex-col justify-between">
+    <div className="h-full min-w-[3.5rem] bg-[#232135] flex flex-col justify-between z-20">
       <div className="flex flex-col">
-        <MenuButton icon={Files} isActive />
+        <MenuButton icon={Files} isActive onClick={toggleExplorerVisibility} />
```

Comment:
Today, the `MenuButton` component is always a `div`. As you are assigning an `onClick` handler, the MenuButton should be a `button`. As we only have one button and the others aren't clickable, maybe you could do a simple check inside the `MenuButton` component for now, like:
```js
const Comp = props.onClick ? 'button' : 'div';

return <Comp ... />
```
But this is the perfect case to use https://www.radix-ui.com/docs/primitives/utilities/slot